streamlit_folium/__init__.py

```diff
@@ -114,6 +114,16 @@ def st_folium(
 
     leaflet = generate_leaflet_string(fig)
 
+    children = list(fig.get_root()._children.values())
```

Comment:
Ideally this should probably be expanded to support html children of the map as well, but for this example, just siblings are sufficient, since the legend is inserted as a sibling of the map, directly under the root Figure.

Reply:
I'll take your word for it :)